packages/form-builder/src/components/onboarding/steps/index.tsx

```diff
@@ -0,0 +1,113 @@
+import {__} from "@wordpress/i18n";
+import {compose} from "@wordpress/compose";
+import withButtons from "./withButtons";
+import withText from "./withText";
+import withDefaults from "./withDefaults";
+
+type Placement = 'top'|'top-start'|'top-end'|'bottom'|'bottom-start'|'bottom-end'|'right'|'right-start'|'right-end'|'left'|'left-start'|'left-end';
+
+export default Object.values(compose(
+    withText,
+    withButtons,
+    withDefaults({
+        canClickTarget: false,
+        scrollTo: false,
+        cancelIcon: {
+            enabled: false,
+        },
+        arrow: false,
+    }),
+)([
+    {
+        id: 'welcome',
+        title: __('Welcome to the visual donation form builder!', 'give'),
+        text: __('The following is a quick (less than a minute) tour of the visual donation form builder, to introduce the tools for creating engaging donation forms.', 'give'),
+    },
+    {
+        id: 'canvas',
+        attachTo: { element: '#FormBlocksCanvas', on: 'right-start' as Placement },
+        title: __('Canvas', 'give'),
+        text: __('Add, reorder, and edit blocks and sections here to make up your form.', 'give'),
+    },
+    {
+        id: 'section',
+        attachTo: { element: '.block-editor-block-list__layout .wp-block > div', on: 'right-start' as Placement },
+        title: __('Form Sections', 'give'),
+        text: __('Forms are broken into sections, which contain blocks for fields or content.', 'give'),
+        modalOverlayOpeningRadius: 5, // Match the border radius of the section block element
+    },
+    {
+        id: 'addButton',
+        attachTo: { element: '#AddBlockButtonContainer', on: 'bottom' as Placement },
+        title: __('Sidebar Toggles', 'give'),
+        text: __('These two buttons give you the ability to add and reorder sections and blocks to the canvas, with drag-and-drop ease.', 'give'),
+    },
+    {
+        id: 'addBlock',
+        attachTo: { element: '.givewp-next-gen-sidebar-secondary', on: 'right-start' as Placement },
```

Comment:
One convention I like to use for these types of things is adding classnames with a `js-` prefix so I know that the element is being controlled by js externally and not to change that classname - we might consider using that here.

Reply:
This is a layered application of JavaScript, though, which is why it is targeting an existing class name.

Reply:
yeah i'm really just thinking this classname & others will probably end up changing (for example: to remove the word `next-gen`) & if we update a gutenberg package and the underlying classname we're targeting changes then this stops working.  Just something to think about for longevity

Reply:
Sure. I switched these to target IDs instead.